lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -7858,7 +7858,18 @@ namespace Js
     template <typename T>
     JavascriptString* JavascriptArray::ToLocaleString(T* arr, ScriptContext* scriptContext)
     {
-        uint32 length = ItemTrace<T>::GetLength(arr, scriptContext);
+        uint32 length = 0;
+        if (Js::TypedArrayBase::Is(arr))
+        {
+            // For a TypedArray use the actual length of the array.
+            length = (static_cast<Js::ArrayObject*>(arr))->GetLength();
```

Comment:
The actual parameter arr has the type Js::RecyclableObject. We have to explicitly call the ArrayObject::GetLength to get the actual length of the array.